game/engine.py

```
@@ -834,7 +834,12 @@ def _load_opening_book(cls) -> dict:
         if cls._opening_book is None:
             try:
                 with open(cls.OPENING_BOOK_PATH, encoding='utf-8') as fh:
-                    cls._opening_book = json.load(fh)
+                    data = json.load(fh)
+
+                if not isinstance(data, dict):
+                    raise ValueError("Opening book must be a dict of FEN -> moves")
```

Ruff (0.15.14)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)